### PR TITLE
Tickets/dm 35454

### PR DESCRIPTION
**Pull request by teweitsai**
* Update the conda recipe for multiple versions of python.
* Ignore the error in `tcpip.close_stream_writer()`.

### CODE REVIEW

File: python/lsst/ts/m2/tcp_client.py [hunk 1]
```diff
@@ -248,7 +248,14 @@ async def _basic_close(self):
...
 
         if self.writer is not None:
-            await tcpip.close_stream_writer(self.writer)
+            # Ignore the TimeoutError (or others) from
+            # asyncio/selector_events.py. A related discussion is here:
+            # https://github.com/home-assistant/core/issues/10468
+            try:
+                await tcpip.close_stream_writer(self.writer)
+            except Exception as error:
+                self.log.debug(f"Error in self._basic_close(): {error}")
```
Comment (tribeiro) [inline python/lsst/ts/m2/tcp_client.py:257]: Consider doing the following instead:

```
except Exception:
    self.log.exception("Error closing stream writer. Ignoring.")
```

This way, we will get the fill traceback of the exception.

**Review by tribeiro — DISMISSED** [review]
Looks good but please, see my inline comment about logging the exception instead of using debug.

**Comment by tribeiro** [comment]
So, I was looking more closely and I found some other places where the issue we experienced might occur.

Basically, the problem is that, when we are doing a downward state transition (from `ENABLED` to `DISABLED` or `DISABLE`/`FAULT` to `STANDBY` we want to make sure the transition occurs even if there are problems in the cleanup process. For example, we don't want to have the CSC stuck in `ENABLED` if it cannot disconnect cleanly from the controller. The CSC should still transition and forcibly close/cleanup the connection.

Not that this is different than the behavior we want when doing an upward state transition. If the CSC cannot connect, then it should fail the state transition. In general, if the CSC is in `STANDBY` and it cannot safely transition to `DISABLED` it should reject the command and stay in `STANDBY`. But if the CSC is in `DISABLED` and fails to go to `ENABLED`, in general, it should probably go to `FAULT`.

That being said, please, take a look at the downward state transition method and make sure you shield them from errors. For example, there are several places in [do_standby](https://github.com/lsst-ts/ts_m2/blob/03679383c7596fdc3a6de0859b91e99baa1c5235/python/lsst/ts/m2/csc.py#L515-L539) that can potentially fail. I think you could encapsulate all that in a try/except clause and make sure you ignore potential errors. 

I mentioned `do_standby` but there are other state transition methods that could benefit from a similar shielding.

File: python/lsst/ts/m2/tcp_client.py (hunk 1 above)
Comment (teweitsai) [reply python/lsst/ts/m2/tcp_client.py]: Updated in the commit: [974f955](https://github.com/lsst-ts/ts_m2/pull/27/commits/974f9550581173e28bc58ecf541a3cc7f3a07029). Thanks!

**Comment by teweitsai** [comment]
I think what will happen now is like this:
1. The CSC is under the `Enabled` state now.
2. For somehow, the TCP/IP connection fails in OS or someone kills the M2 EUI at VM.
3. The user wants to transition the CSC into the `Standby` state.
4. The `OSError` will be triggered because of no connection, and put the system into the `Fault` state. Based on the Bruno's result, you can see the `OSError` was triggered, and the problem came from the OS to close the stream writer. After this fix, this error will be ignored and the CSC will transition to the `Fault` state.
5. Then, the CSC can transition to the `Standby` state without the problem because the `Model` had already closed all the connections when dealing with the `OSError`.

This should solve the problem that the CSC might stuck at the `Enabled` state. I think it might make things complex if we have the different behaviors between the transition-up and transition-down. Instead, when there is something wrong, we just put the CSC into the `Fault` state, and the user will notice this. The bottom line is that the user can transition to the `Standby` state from the `Fault` state. Thanks!

**Review by tribeiro — APPROVED** [review]

**Pull request merged**